internal/service/gateway/gateway.go

```diff
@@ -34,11 +34,13 @@ const (
 
 func ResourceGateway() *schema.Resource {
 	return &schema.Resource{
-		Description:   "Network gateways connect SDN Private Networks to external IP networks.",
-		CreateContext: resourceGatewayCreate,
-		ReadContext:   resourceGatewayRead,
-		UpdateContext: resourceGatewayUpdate,
-		DeleteContext: resourceGatewayDelete,
+		EnableLegacyTypeSystemApplyErrors: true,
```

Comment:
Probably better to remove the commit that adds these `EnableLegacyTypeSystem*Errors` for now and look into re-implementing this resource with plugin framework 🤔